sycl/test-e2e/Basic/group_async_copy.cpp

```diff
@@ -122,21 +122,21 @@ template <typename T> int test(size_t Stride) {
        size_t Offset = GrId * WorkGroupSize;
        if (Stride == 1) { // Check the version without stride arg.
          auto E = NDId.async_work_group_copy(
-             local_ptr<T>(Local), In.get_pointer() + Offset, NElemsToCopy);
+             local_ptr<T>(Local), global_ptr<T>(In) + Offset, NElemsToCopy);
```

Comment:
Could we use `get_multi_ptr()` instead? I would hope we could use `decorated::no`, but some of the tests may need it to be `legacy`.

Reply:
Thanks @steffenlarsen 
Happy to update.
The main reasons I did not use get_multi_ptr() were i) using globa_ptr conformed to the type get_pointer used to return and ii) the ambiguities  involved where to use which decorated type, iii) (which is not important) numerous use of unpleasant syntax like in.template get_multi_ptr<accessdecorated::legacy>

Reply:
> 1- Modifying `global_ptr` aliasing to use `decorated::no`, and keep use of `global_ptr` when it's right

Do you mean specify the decoration in the use of `global_ptr` in the tests? I.e. use `global_ptr<T, sycl::decorated::no>`? Or do you mean change the actual alias in the SYCL headers to use `sycl::decorated::no`?

- If the former: Sure, I am okay with that, but I don't see why it is much better than using `get_multi_ptr`? It means you won't need to specify the underlying type.
- If the latter: We cannot do that. The SYCL specification defines it as defaulting to `legacy` for compatibility with SYCL 1.2.1. We can use `raw_global_ptr` though, which I would be okay with.

> 2- Use `get_multi_ptr<decorated::legacy>` in places where `mulit_ptr<decorated::legacy>` is needed.

Sure. That sounds fine to me.